lms/static/scripts/frontend_apps/components/test/BasicLtiLaunchApp-test.js

```diff
@@ -433,6 +455,71 @@ describe('BasicLtiLaunchApp', () => {
       await waitForElement(wrapper, 'FakeDialog[title="Authorize Hypothesis"]');
       await contentHidden(wrapper);
     });
+
+    context('when auth fails multiple times in a row', () => {
+      // Helper to ensure the dialog is initially present but also remains so
+      // for a short duration longer without disappearing.
+      async function dialogShouldRemain(wrapper) {
+        // Add an artificial delay to ensure that all the api calls resolve or
+        // reject before proceeding.
+        await true;
+        await true;
```

Comment:
This is not really safe as it assumes that API calls will take a certain number of microtasks before being resolved. In my experience this can make tests brittle and is liable to cause unexpected failures with apparently innocent refactoring.

Having a think about the original issue, the problem was that the auth dialog would be hidden and the iframe shown briefly when the LMS app was in the following state:

- User has clicked the "Authorize" button
- API calls to the backend to fetch content URL or fetch groups have not yet completed

In this branch, the dialog remains visible until both those calls complete. Therefore what I would suggest to do in the tests is instead of making the content URL / groups API calls reject, instead make them not resolve immediately and then check whether the dialog is visible while the API calls are "in flight". Then resolve the API calls after this assert passes.

Reply:
sounds reasonable, I'll give it a go!